commit 143dfe7f307e4f5c75cfa01c978b2c72a6c024d3 (refs/changes/72/2772/2)
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2017-10-11T14:43:30-04:00 (2 years ago)
    
    HEAD-2369 cache=false with sdc-server setup still creates a cache device

diff --git a/scripts/joysetup.sh b/scripts/joysetup.sh
index 97b83d34..2b4e6cfb 100755
--- a/scripts/joysetup.sh
+++ b/scripts/joysetup.sh
@@ -625,7 +625,8 @@ if [[ "$(zpool list)" == "no pools available" ]] \
         dl3="${CONFIG_layout}"
     fi
 
-    if ! /usr/bin/disklayout ${dl0} "${dl1}" ${dl2} "${dl3}"> ${POOL_FILE}; then
+    if ! /usr/bin/disklayout ${dl0} ${dl1:+"$dl1"} ${dl2} "${dl3}" \
+         > ${POOL_FILE}; then
         fatal "disk layout failed"
     fi
 
